src/local/butler/run_bot.py

```diff
@@ -98,7 +98,9 @@ def _setup_environment_and_configs(args, appengine_path):
     os.environ['LOCAL_GCS_BUCKETS_PATH'] = local_gcs_buckets_path
 
   if args.android_serial:
-    os.environ['OS_OVERRIDE'] = 'ANDROID'
+    if os.environ['OS_OVERRIDE'] != 'ANDROID_KERNEL':
```

Comment:
This will exception if OS_OVERRIDE is not set
Probably just 
if not os.getenv('OS_OVERRIDE'):